app/models/katello/pool.rb

```diff
@@ -40,16 +53,26 @@ def self.active(subscriptions)
     end
 
     def self.expiring_soon(subscriptions)
-      subscriptions.select { |s| (s.end_date - Date.today) <= DAYS_EXPIRING_SOON }
+      subscriptions.select { |s| (s.end_date.to_date - Date.today) <= DAYS_EXPIRING_SOON }
     end
 
     def self.recently_expired(subscriptions)
       today_date = Date.today
 
       subscriptions.select do |s|
-        end_date = s.end_date
+        end_date = s.end_date.to_date
         today_date >= end_date && today_date - end_date <= DAYS_RECENTLY_EXPIRED
       end
     end
+
+    def quantity_available
+      self.quantity - self.consumed if self.quantity && self.consumed
```

Comment:
If these are not defined, the function returns nil which is not what I would expect from a function like this. Perhaps default to 0?

```
return 0 unless self.quantity && self.consumed
self.quantity - self.consumed
```